plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java

```diff
@@ -45,13 +59,31 @@ public interface ModelExecutor {
      * @return an ActionRequest instance for remote model inference
      * @throws IllegalArgumentException if the input parameters are null
      */
-    default <T> ActionRequest getRemoteModelInferenceRequest(Map<String, String> parameters, String modelId) {
+    default <T> ActionRequest getRemoteModelInferenceRequest(Map<String, String> parameters, String modelId, String functionName) {
+        MLInput mlInput = new MLInput();
         if (parameters == null) {
             throw new IllegalArgumentException("wrong input. The model input cannot be empty.");
         }
-        RemoteInferenceInputDataSet inputDataSet = RemoteInferenceInputDataSet.builder().parameters(parameters).build();
+        if (functionName.equals("remote")) {
+            RemoteInferenceInputDataSet inputDataSet = RemoteInferenceInputDataSet.builder().parameters(parameters).build();
+            mlInput = MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inputDataSet).build();
+        } else if (functionName.equals("text_embedding") || functionName.equals("sparse_encoding")) {
```

Comment:
Only these two algorithm models? Check if this line can be simplified to FunctionName.isDLModel(functionName)